ERCS/erc-7920.md

```diff
@@ -0,0 +1,358 @@
+```
+
+Result:
+
+```JavaScript
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "result": {
+    "signature": "0x4355c47d63924e8a72e509b65029052eb6c299d53a04e167c5775fd466751c9d07299936d304c153f6443dfa05f40ff007d72911b6f72307f996231605b915621c",
+    "merkleRoot": "0x7de103665e21d6c9d9f82ae59675443bd895ed42b571c7f952c2fdc1a5b6e8d2",
+    "proofs": [
+      ["0x4bdbac3830d492ac3f4b0ef674786940fb33481b32392e88edafd45d507429f2"],
+      ["0x95be87f8abefcddc8116061a06b18906f32298a4644882d06baff852164858c6"]
+    ]
+  }
+}
+```
+
+## Rationale
+
+### UX Improvement
+
+A single signature that covers multiple messages
+
+### Isolated Verification
+
+Independent verification of messages without knowledge of others
+
+### Human-readable
+
+This ERC preserves the readability benefits of EIP-712. Giving wallets and users insight into what is being signed.
+
+### Efficient verification on-chain
+
+`_verifyMerkleProof` has a runtime of `O(log2(N))` where N is the number of messages that were signed.
+
+When N=1, merkleRoot **must** equal keccak256(encode(m₁)) and proofs MUST be empty arrays
+
+### Improved Wallet Security
+
+Certain messages signed in isolation may appear harmless but combined with may be harmful. Giving the full list of messages to users could help them better navigate their experience.
+
+### Flexible Verification Modes
+
+Applications can require combination of messages be signed together to enhance security.
+
+## Backwards Compatibility
+
+When the number of message is one, `eth_signTypedData_v5` produces the same signature as `eth_signTypedData_v4` since `merkleRoot == keccak256(encode(message))`. This allows `eth_signTypedData_v5` to be a drop-in replacement for `eth_signTypedData_v4` with no changes to on-chain verification.
+
+## Reference Implementation
+
+Reference implementation of `eth_signTypedData_v5` can be found the assets directory.
```

Comment:
You can and should link directly to the file (something like `../assets/erc-7920/eth_signTypedData_v5.ts`)